src/linear_algebra/projective_space/basic.lean

```diff
@@ -0,0 +1,232 @@
+/-
+Copyright (c) 2022 Adam Topaz. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Adam Topaz
+-/
+import linear_algebra.finite_dimensional
+
+/-!
+
+# Projective Spaces
+
+This file contains the definition of the projectivization of a vector space over a field,
+as well as the bijection between said projectivization and the collection of all one
+dimensional subspaces of the vector space.
+
+## Constructing terms of `projectivization K V`.
+We have three ways to construct terms of `projectivization K V`:
+- `projectivization.mk K v hv` where `v : V` and `hv : v ≠ 0`.
+- `projectivization.mk' K v` where `v : { w : V // w ≠ 0 }`.
+- `projectivization.mk'' H h` where `H : submodule K V` and `h : finrank H = 1`.
+
+## Other definitions
+- For `v : projectivization K V`, `v.submodule` gives the corresponding submodule of `V`.
+- `projectivization.equiv_submodule` is the equivalence between `projectivization K V`
+  and `{ H : submodule K V // finrank H = 1 }`.
+- For `v : projectivization K V`, `v.rep : V` is a representative of `v`.
+
+-/
+
+variables (K V : Type*) [field K] [add_comm_group V] [module K V]
+
+/-- The setoid whose quotient is the projectivization of `V`. -/
+def projectivization_setoid : setoid { v : V // v ≠ 0 } :=
+{ r := λ u v, ∃ a : K, a • (u : V) = v,
+  iseqv := begin
+    refine ⟨λ u, ⟨1, by simp⟩, _, λ u v w ⟨a,ha⟩ ⟨b,hb⟩, ⟨b * a, by simp [ha, hb, mul_smul]⟩⟩,
+    rintros u v ⟨a,ha⟩,
+    use a⁻¹,
+    rw [← ha, ← mul_smul, inv_mul_cancel, one_smul],
+    intros c, apply v.2,
+    simpa [c] using ha.symm,
+  end }
+
+
+/-- The projectivization of the `K`-vector space `V`. -/
+def projectivization := quotient (projectivization_setoid K V)
+
+namespace projectivization
+
+variables {V}
+
+/-- Construct an element of the projectivization from a nonzero vector. -/
+def mk (v : V) (hv : v ≠ 0) : projectivization K V := quotient.mk' ⟨v,hv⟩
+
+/-- A variant of `projectivization.mk` in terms of a subtype. `mk` is preferred. -/
+def mk' (v : { v : V // v ≠ 0 }) : projectivization K V := quotient.mk' v
+
+@[simp] lemma mk'_eq_mk (v : { v : V // v ≠ 0}) :
+  mk' K v = mk K v v.2 :=
+by { dsimp [mk, mk'], congr' 1, simp }
+
+noncomputable instance [nontrivial V] : inhabited (projectivization K V) :=
+let e := exists_ne (0 : V) in ⟨mk K e.some e.some_spec⟩
+
+variable {K}
+
+/-- Choose a representative of `v : projectivization K V` in `V`. -/
+protected noncomputable def rep (v : projectivization K V) : V := v.out'.1
+
+lemma rep_nonzero (v : projectivization K V) : v.rep ≠ 0 := v.out'.2
+
+@[simp]
+lemma mk_rep (v : projectivization K V) :
+  mk K v.rep v.rep_nonzero = v :=
+by { dsimp [mk, projectivization.rep], simp }
+
+open finite_dimensional
+
+/-- Consider an element of the projectivization as a submodule of `V`. -/
+protected def submodule (v : projectivization K V) : submodule K V := K ∙ v.rep
```

Comment:
If you have the energy, I think it would be more elegant to construct this using [quotient.lift](https://leanprover-community.github.io/mathlib_docs/init/data/quot.html#quotient.lift).

Reply:
I agree: https://github.com/leanprover-community/mathlib/pull/12438#discussion_r819121696

Reply:
done!